conference.js

```diff
@@ -875,14 +890,28 @@ export default {
         this.invite = new Invite(room);
         this._room = room; // FIXME do not use this
 
-        let email = APP.settings.getEmail();
-        email && sendData(this.commands.defaults.EMAIL, email);
+        const localParticipantJoinedOptions = {};
+        const email = APP.settings.getEmail();
+        if (email) {
+            sendData(this.commands.defaults.EMAIL, email);
+            localParticipantJoinedOptions.email = email;
+        }
+
+        const avatarUrl = APP.settings.getAvatarUrl();
+        if (avatarUrl) {
+            sendData(this.commands.defaults.AVATAR_URL, avatarUrl);
+            localParticipantJoinedOptions.avatarUrl = avatarUrl;
+        }
+
+        const avatarId = APP.settings.getAvatarId();
+        localParticipantJoinedOptions.avatarId = avatarId;
+        if(!email && !avatarUrl) {
+            sendData(this.commands.defaults.AVATAR_ID, avatarId);
+        }
+
+        APP.store.dispatch(
```

Comment:
Do we really need to carry all three of those throughout the code? email, avatar_url and avatar_id. Shouldn't we move the generation and reading part from Settings so that we don't need to carry those separately from the participant and the avatar objects?